server/controllers/createIncident/createIncident.ts

```diff
@@ -0,0 +1,81 @@
+import { NextFunction, Response } from 'express'
+import FormWizard from 'hmpo-form-wizard'
+import backUrl from '../../utils/backUrl'
+import FormInitialStep from '../base/formInitialStep'
+import { type NewIncident } from '../../data/incidentReportingApi'
+
+export default class CreateIncident extends FormInitialStep {
+  middlewareSetup() {
+    super.middlewareSetup()
+    this.use(this.setOptions)
+  }
+
+  async setOptions(req: FormWizard.Request, res: Response, next: NextFunction) {
+    const { prisonApi } = res.locals.apis
+    const prisonsLookup = await prisonApi.getPrisons()
+
+    req.form.options.fields.prisonId.items = Object.values(prisonsLookup).map(prison => ({
+      value: prison.agencyId,
+      text: prison.description,
+    }))
+
+    next()
+  }
+
+  locals(req: FormWizard.Request, res: Response): object {
+    const locals = super.locals(req, res)
+
+    const backLink = backUrl(req, {
+      fallbackUrl: '/incidents/',
+    })
+
+    return {
+      ...locals,
+      backLink,
+      cancelLink: backLink,
+    }
+  }
+
+  async saveValues(req: FormWizard.Request, res: Response, next: NextFunction) {
+    try {
+      const { incidentType } = req.form.values
+      const { incidentDate } = req.form.values
+      const { incidentTime } = req.form.values
```

Comment:
not a biggie, but you could probably destructure all of these at once, e.g.

```JavaScript
const { incidentType, incidentDate, incidentTime, incidentTitle, incidentPrisonId } = req.form.values
```

Reply:
good point, changed these now